api/v1/controllers/collectors.js

```diff
@@ -44,7 +44,7 @@ const MINUS_ONE = -1;
  * attribute.
  * @param  {String}   authToken - Collector authentication token
  * @param  {String}   timestamp - Timestamp sent by collector in heartbeat
- * @returns {Object} Sample generator with reencrypted context values.
+ * @returns {Promise<Generator>} Sample generator with reencrypted context values.
```

Comment:
Yessss! This is a much better way of documenting the expected return type.

Reply:
Java 5 :)